src/components/App/Wrapper.js

```diff
@@ -1,10 +1,44 @@
-import styled from 'styled-components';
+import React, { useEffect, useState } from 'react';
+import styled, { withTheme } from 'styled-components';
+import { animated, useTransition, config } from 'react-spring';
+import { colorLightTeal } from '../../styles/designTokens';
+import extractRoleFromPath from '../../utils/extractRoleFromPath';
+import generateColors from '../../utils/generateColors';
 
-const Wrapper = styled.div`
+const Wrapper = styled(animated.div)`
+  /* TODO */
+  /* We need positon: absolute in that wrapper for proper container positionin during animation */
   min-height: 100%;
   display: grid;
   grid-template-rows: auto 1fr auto;
-  background-color: ${props => props.theme.primaryColor};
 `;
 
-export default Wrapper;
+function Animator({ children, location, theme }) {
+  const { pathname } = location;
+  const { primaryColor } = theme;
+  const currentRole = extractRoleFromPath(pathname);
+
+  const [previousRole, setPreviousRole] = useState(currentRole);
+  useEffect(() => setPreviousRole(currentRole), [previousRole, currentRole]);
```

Comment:
I think in **deps array** we don't need to specify **previousRole**, since it is not used 'inside' effect and won't trigger change anyway.

Reply:
:+1: leftover from experiments.